source/Optimizers/nelder_mead.d

```diff
@@ -331,7 +320,7 @@ unittest {
         assert(sphere(new Vector!real([0.0])) == 0.0);
         assert(sphere(new Vector!real([1.0, -1.0])) == 2.0);
 
-        size_t num = 100;
+        size_t num = 25;
         real succes = 0;
 
         foreach(i; 0 .. num)

```

Comment:
How much time does this take? Should it be turned to the integration test function?

Sometimes all we need is the unit test to check the input/output contract is valid (so just one iteration with different hyperparameters) and checking that it converges is left to end-to-end tests/integration tests (so we can unittest in <10seconds)

Reply:
Too much times..
I agree with your idea, we should make sure the right modification is applied (contraction/..) and that everything is rightly computed. Any convergence test should go in the integration tests.

We can do it after this commit as it was related to the bug in the core loop.
After that, you could push the changes you made in random_search and full_tests.d together with better unittests for random_search.